scrapy_autoextract/providers.py

```diff
@@ -45,17 +46,56 @@ def __init__(
         self.stats = stats
         self.settings = settings
 
+    def __before__(self):
+        providers = self.request.meta.get("_autoextract_providers", [])
+        self.request.meta["_autoextract_providers"] = providers + [self]
+
     async def __call__(self):
         """Make an AutoExtract request and build a Page Input of provided class
         based on API response data.
         """
-        page_type = self.get_page_type()
-        self.stats.inc_value(f"autoextract/{page_type}/total")
+        data = await self._fetch_data()
+        if "error" in data:
+            self.stats.inc_value("autoextract/error/query")
+            raise QueryError(data["query"], data["error"])
+
+        return self.provided_class(data=data)
+
+    @classmethod
+    def register(cls):
+        """Register this provider for its provided class on scrapy-poet
+        registry. This will make it possible to declare provided class as
+        a callback dependency when writing Scrapy spiders.
+        """
+        register(cls, cls.provided_class)
+
+    async def _fetch_data(self):
+        page_type = self.provided_class.item_key
+
+        # When the backend includes support for multiple page types in a single
+        # request, we'll be able to remove this condition and always try to
+        # return the cached data on request's meta.
+        if page_type == "html":
+            # Retrieve cached data on request's meta
+            data = self.request.meta.get("_autoextract_data")
+            if data is not None:
+                return data
+
+        # When backend includes support for "html" page type, we'll be able to
+        # remove this condition.
+        if page_type == "html":
+            # There's no support on the backed for an "html" page type yet,
+            # so let's use a dummy page type here
+            page_type = "article"
```

Comment:
This is a temporary hack to make HTML dependency work alone without the need to combine it with a Product or Article dependency, for instance. We're using a dummy page type here whilst AutoExtract backend doesn't support this "html" one.